src/args.rs

```diff
@@ -779,6 +782,33 @@ impl<'a> ArgMatches<'a> {
         btypes.build().map_err(From::from)
     }
 
+    /// Parses the max-filesize argument option into a byte count.
+    fn max_filesize(&self) -> Result<Option<u64>> {
```

Comment:
This duplication isn't acceptable, sorry.

Please just get rid of the clap validator. The error can be reported from here. That will also get rid of the build dep on regex.